src/client/datascience/notebook/notebookControllerManager.ts

```diff
@@ -377,6 +378,24 @@ export class NotebookControllerManager implements INotebookControllerManager, IE
         }
     }
 
+    private handlerAddedForRemoteKernels?: boolean;
+    private reloadControllersAfterChangingRemote() {
+        if (this.handlerAddedForRemoteKernels) {
+            return;
+        }
+        this.handlerAddedForRemoteKernels = true;
+        this.serverUriStorage.onDidChangeUri(async () => {
```

Comment:
When remote URI changes refresh the kernels.
Will need to add more smarts to refresh this when we start a start/stop kernel as well (so that we get teh live kernels showing up)